src/util/Vocabularies.ts

```diff
@@ -2,57 +2,81 @@
 
 /**
  * Creates a function that expands local names from the given base URI,
  * and exports the given local names as properties on the returned object.
  */
-export function createNamespace<TKey extends string, TValue>(
-  baseUri: string,
-  toValue: (expanded: string) => TValue,
-  ...localNames: TKey[]):
-  Namespace<typeof localNames, TValue> {
-  const expanded: Namespace<typeof localNames, TValue> = {} as any;
-  // Expose the main namespace
-  expanded.namespace = toValue(baseUri);
+export function createNamespace<TBase extends string, TLocal extends string>(baseUri: TBase, localNames: TLocal[]):
+Namespace<TBase, TLocal> {
+  const expanded: ExpandedRecord<TBase, TLocal> = { } as any;
   // Expose the listed local names as properties
   for (const localName of localNames) {
-    (expanded as RecordOf<TKey[], TValue>)[localName] = toValue(`${baseUri}${localName}`);
+    expanded[localName] = `${baseUri}${localName}`;
   }
-  return expanded;
-}
-
-/**
- * Creates a function that expands local names from the given base URI into strings,
- * and exports the given local names as properties on the returned object.
- */
-export function createUriNamespace<T extends string>(baseUri: string, ...localNames: T[]):
-Namespace<typeof localNames, string> {
-  return createNamespace(baseUri, (expanded): string => expanded, ...localNames);
+  return {
+    namespace: baseUri,
+    ...expanded,
+  };
 }
 
 /**
  * Creates a function that expands local names from the given base URI into named nodes,
  * and exports the given local names as properties on the returned object.
  */
-export function createTermNamespace<T extends string>(baseUri: string, ...localNames: T[]):
-Namespace<typeof localNames, NamedNode> {
-  return createNamespace(baseUri, DataFactory.namedNode, ...localNames);
+export function createTermNamespace<TBase extends string, TLocal extends string>(namespace: Namespace<TBase, TLocal>):
+TermNamespace<Namespace<TBase, TLocal>> {
+  // Need to cast since `fromEntries` typings aren't strict enough
+  return Object.fromEntries(
+    Object.entries(namespace).map(([ key, value ]): [string, NamedNode] => [ key, DataFactory.namedNode(value) ]),
+  ) as TermNamespace<Namespace<TBase, TLocal>>;
 }
 
 /**
  * Creates a function that expands local names from the given base URI into string,
  * and exports the given local names as properties on the returned object.
  * Under the `terms` property, it exposes the expanded local names as named nodes.
  */
-export function createUriAndTermNamespace<T extends string>(baseUri: string, ...localNames: T[]):
-Namespace<typeof localNames, string> & { terms: Namespace<typeof localNames, NamedNode> } {
-  return Object.assign(createUriNamespace(baseUri, ...localNames),
-    { terms: createTermNamespace(baseUri, ...localNames) });
+export function createUriAndTermNamespace<TBase extends string, TLocal extends string>(baseUri: TBase,
```

Comment:
I wonder if, at this point, we don't just want to export only this last function and call it `createNamespace` or `createVocabulary`, but up to you. (The other internal function could be `createValueNamespace`.)

Reply:
The reason I didn't to it was to make the impact as small as possible. Same reason `namespace` is still used above instead of something like `base`. Although this still is a breaking change since I changed the input of the `createNamespace` function so perhaps I should go all out with the changes since it's breaking anyway.

Reply:
Exactly.

Reply:
Going to change the function to `createVocabulary`, change the `Namespace` and `TermNamespace` interfaces to `ValueVocabulary` and `TermVocabulary` (but not export them), and keep the `namespace` field. As that seems to be most correct regarding naming.

One thing I realized is that the typings do cause issues when calling the function with a dynamic array. E.g.:
```ts
const values: string[] = [ 'test' ];
const test = createVocabulary('http://example.com/', ...values);
```
`test.whatever` now has type `http://example.com/${string}` 😅 . Will see if there is somehow a way to detect that the input is not a string union type (although my guess is no) and return a different type based on that. Not a big issue since we don't use it like that but would be nice to be fully correct.

Reply:
Turns out is is possible. It's not possible to have the fancy template literal typings in that case (or at least I didn't find how to do it). So in the example above `test.whatever` would now have type `string | undefined`. Which I guess is good enough as it at least allows that use case to still be used.